improver_tests/utilities/test_OccurrenceWithinVicinity.py

```diff
@@ -395,7 +394,16 @@ def test_no_realization_or_time(request, cube_with_realizations, land_fixture):
     cube = cube[0]
     cube.data[2, 1] = 1.0
     orig_shape = cube.data.shape
-    result = OccurrenceWithinVicinity(DISTANCE, land)(cube)
+    result = OccurrenceWithinVicinity(radius=RADIUS, land_mask_cube=land)(cube)
     assert isinstance(result, Cube)
     assert result.data.shape == orig_shape
     assert np.allclose(result.data, expected)
+
+
+def test_two_radii_provided_exception(cube):
+    """Test an exception is raised if both radius and grid_point_radius are
+    provided as arguments."""
+    with pytest.raises(
+        ValueError, match="Only one of radius or grid_point_radius should be set"
+    ):
+        OccurrenceWithinVicinity(radius=2000, grid_point_radius=2)
```

Comment:
If I specify `radius=2000, grid_point_radius=0`, the code will silently ignore `grid_point_radius`. Admittedly, this is user-error, but it isn't clear what I asked for, so the code should fail.

Reply:
Check in the code extended to include `is not None`. This now picks up this case which I've added by parameterizing the test to try a couple of radii, one of which is 0.